lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -234,15 +234,8 @@ module InternalConstants
 
     PLUGIN_NAME = 'Fluentd Google Cloud Logging plugin'.freeze
     # Extract plugin version by finding the spec this file was loaded from.
-    PLUGIN_VERSION = begin
-      dependency = Gem::Dependency.new('fluent-plugin-google-cloud')
-      all_specs, = Gem::SpecFetcher.fetcher.spec_for_dependency(dependency)
-      matching_spec, = all_specs.grep(
-        proc { |spec,| __FILE__.include?(spec.full_gem_path) }) do |spec,|
-          spec.version.to_s
-        end
-      matching_spec
-    end.freeze
+    PLUGIN_VERSION = __FILE__.match(
+      %r{fluent-plugin-google-cloud-?(?<version>[0-9a-zA-Z\.]*)/})['version']
```

Comment:
Note that when unit tests run, the `__FILE__` path does not have version embedded. It's simply `.../fluent-plugin-google-cloud/lib/fluent...`. So the regex has to be compatible to that.

Reply:
Does this mean that in local tests we will have an empty version? Can we keep the original code as a fallback?

Reply:
It will not affect testing a gem locally. When installing a gem, it will be installed under the expected file path. It only affects unit tests. Is that something we are concerned with?

Reply:
I think it makes sense to ensure that the plugin version always comes from the gem config, even when testing a local clone without an installed gem. WDYT?

Reply:
Updated. PTAL